test/can_obd2/can_mapping_test.cpp

```diff
@@ -72,85 +76,128 @@ void CANMappingTest::formula_test(void)
+                                }
+
+                                /* shift it all the way to the left */
+                                uint64_t shifted_test_value = test_value << (64-length);
+
+                                for (uint8_t offset = 0; offset < (CAN_MSG_SIZE * 8) - length + 1; offset++) {
+                                        CAN_msg msg;
+                                        CANMapping mapping;
+                                        memset(&mapping, 0, sizeof(mapping));
+                                        memset(&msg, 0, sizeof(CAN_msg));
+                                        mapping.offset = offset;
+                                        mapping.length = length;
+                                        mapping.type = CANMappingType_unsigned;
+                                        mapping.bit_mode = true;
+                                        mapping.big_endian = (bool)endian;
+
+                                        /* now shift the value by the offset */
+                                        uint64_t offset_test_value = shifted_test_value >> offset;
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+                                        /* account for platform's endian-ness */
+                                        offset_test_value= swap_uint64(offset_test_value);
+#endif
+                                        msg.data64 = offset_test_value;
 
-#ifdef CAN_MAPPING_TEST_DEBUG
-                    printf("bitmode test: value / offset / length / return %lu %d %d %f\r\n" ,
-                           test_value, offset, length, value);
-                    printf("CAN data: ");
-                    for (size_t di = 0; di < CAN_MSG_SIZE; di++){
-                    		uint8_t d = msg.data[di];
-                            printf("%02x ", d);
-                    }
-                    printf("\r\n");
-
-                    for (size_t di = 0; di < CAN_MSG_SIZE; di++){
-                    		uint8_t d = msg.data[di];
-                            static char b[9];
-							b[0] = '\0';
-							for (int z = 128; z > 0; z >>= 1)
-							{
-								strcat(b, ((d & z) == z) ? "1" : "0");
-							}
-                            printf("%s ", b);
-                    }
-                    printf("\r\n");
+                                        float value = canmapping_extract_value(msg.data64, &mapping);
 
+                                        /* prepare the comparison value */
+                                        uint64_t compare_value = test_value;
+                                        if (!mapping.big_endian) {
+                                                compare_value = swap_uint_length(test_value, length);
+                                        }
+#ifdef CAN_MAPPING_TEST_DEBUG
+                                        printf("bitmode test: endian=%u / test_value=%lu / offset=%d / length=%d / return = %f\r\n" ,
```

Comment:
I've wanted to keep that for development purposes if/when it needs to be debug'd. So, I prefer keeping it.